### DISCUSSION TITLE
Do timestamps get implicitly casted to another timezone?

### DISCUSSION THREAD

**Discussion by Abdullahsab3** [apache/datafusion#10311] [Q&A] [answered]
While I was looking into https://github.com/apache/datafusion/issues/10308 , I noticed in this query: 
```
select 
    date_bin(interval '1 hour', time at time zone 'Europe/Brussels') 
from 
    raw_data 
where 
    time >= '2021-03-27T22:00:00.000Z' 
and 
    time <= '2021-03-29T00:00:00.000Z'
-- exclude the daylight savings hour
and time not between '2021-03-28T02:00:00Z' and '2021-03-28T03:00:00Z'
group by date_bin(interval '1 hour', time at time zone 'Europe/Brussels') 
```
that the timestamps in the `where` clause seem to be interpreted as a timestamp with 'Europe/Brussels' timezone, which means that the `Z` seems to be ignored. The `and time not between '2021-03-28T02:00:00Z' and '2021-03-28T03:00:00Z'` clause, for instance, refers to the daylights saving hour in local time, while it should be referring to the hour between  2021-03-28T02:00:00 UTC and 2021-03-28 03:00:00 UTC.

Maybe I misunderstood something there 🤔

**Comment by alamb** [comment, marked as answer]
I think the timestamps may be interpreted as "None" which then has special rules for casting

You can see the actual types using the [arrow_typeof](https://datafusion.apache.org/user-guide/sql/scalar_functions.html#arrow-typeof) function


```sql
> select '2021-03-27T22:00:00.000Z'::timestamp;
+----------------------------------+
| Utf8("2021-03-27T22:00:00.000Z") |
+----------------------------------+
| 2021-03-27T22:00:00              |
+----------------------------------+
1 row(s) fetched.
Elapsed 0.011 seconds.

> select arrow_typeof('2021-03-27T22:00:00.000Z'::timestamp);
+------------------------------------------------+
| arrow_typeof(Utf8("2021-03-27T22:00:00.000Z")) |
+------------------------------------------------+
| Timestamp(Nanosecond, None)                    |
+------------------------------------------------+
1 row(s) fetched.
Elapsed 0.002 seconds.
```

**Comment by alamb** [comment]
Filed https://github.com/apache/datafusion/issues/10344 to try and document the current behavior
